I was trying to build for the hammerhead device. These are the errors I got before this patch:
```
In file included from hardware/qcom/display/msm8974/libmemtrack/kgsl.c:23:
In file included from hardware/libhardware/include/hardware/memtrack.h:24:
hardware/libhardware/include/hardware/hardware.h:23:10: fatal error: 'cutils/native_handle.h' file not found
#include <cutils/native_handle.h>
         ^~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
[  0% 814/92379] target thumb C: memtrack.msm8974 <= hardware/qcom/display/msm8974/libmemtrack/memtrack_msm.c
FAILED: out/target/product/hammerhead/obj/SHARED_LIBRARIES/memtrack.msm8974_intermediates/memtrack_msm.o
/bin/bash -c "PWD=/proc/self/cwd /usr/bin/ccache prebuilts/clang/host/linux-x86/clang-r487747c/bin/clang -I hardware/libhardware/include -I hardware/qcom/display/msm8974/libmemtrack -I out/target/product/hammerhead/obj/SHARED_LIBRARIES/m
emtrack.msm8974_intermediates -I out/target/product/hammerhead/gen/SHARED_LIBRARIES/memtrack.msm8974_intermediates -isystem device/lge/hammerhead/kernel-headers -isystem device/lge/hammerhead/kernel-headers -Isystem/logging/liblog/includ
e_vndk -Iexternal/libcxx/include -Iexternal/libcxxabi/include -isystem out/soong/.intermediates/bionic/libc/libc/android_vendor.34_arm_armv7-a-neon_shared/gen/include -isystem bionic/libc/kernel/uapi/asm-arm -isystem bionic/libc/kernel/u
api -isystem bionic/libc/kernel/android/scsi -isystem bionic/libc/kernel/android/uapi   -Iexternal/libcxxabi/include    -isystem out/target/product/hammerhead/obj/include -c  -fomit-frame-pointer -DANDROID -fmessage-length=0 -W -Wall -Wn
o-unused -Winit-self -Wpointer-arith -Wunreachable-code-loop-increment -no-canonical-prefixes -DNDEBUG -UDEBUG -fno-exceptions -Wno-multichar -O2 -g -fdebug-default-version=5 -fno-strict-aliasing -Werror=date-time -Werror=pragma-pack -We
rror=pragma-pack-suspicious-include -Werror=string-plus-int -Werror=unreachable-code-loop-increment -Wno-error=deprecated-declarations -D__compiler_offsetof=__builtin_offsetof -faddrsig -fcommon -Werror=int-conversion -Wno-reserved-id-ma
cro -fcolor-diagnostics -Wno-sign-compare -Wno-inconsistent-missing-override -Wno-c99-designator -Wno-gnu-designator -Wno-gnu-folding-constant -Wunguarded-availability -D__ANDROID_UNAVAILABLE_SYMBOLS_ARE_WEAK__ -ffp-contract=off -fdebug-
prefix-map=/proc/self/cwd= -ftrivial-auto-var-init=zero -enable-trivial-auto-var-init-zero-knowing-it-will-be-removed-from-clang -Wno-unused-command-line-argument -Wno-unused-command-line-argument -ffunction-sections -fdata-sections -fno
-short-enums -funwind-tables -fstack-protector-strong -Wa,--noexecstack -D_FORTIFY_SOURCE=2 -Wstrict-aliasing=2 -Werror=return-type -Werror=non-virtual-dtor -Werror=address -Werror=sequence-point -Werror=format-security -nostdlibinc -fde
bug-info-for-profiling -msoft-float -march=armv7-a -mfloat-abi=softfp -mfpu=neon    -std=gnu11 -mthumb -Os   -Wno-error -DDO_NOT_CHECK_MANUAL_BINDER_INTERFACES -D__ANDROID_VNDK__ -D__ANDROID_VENDOR__ -fPIC -DANDROID_STRICT -target armv7a
-linux-androideabi34   -Werror=bool-operation -Werror=format-insufficient-args -Werror=implicit-int-float-conversion -Werror=int-in-bool-context -Werror=int-to-pointer-cast -Werror=pointer-to-int-cast -Werror=xor-used-as-pow -Wno-void-po
inter-to-enum-cast -Wno-void-pointer-to-int-cast -Wno-pointer-to-int-cast -Werror=fortify-source -Werror=address-of-temporary -Werror=null-dereference -Werror=return-type -Wno-tautological-constant-compare -Wno-tautological-type-limit-co
mpare -Wno-reorder-init-list -Wno-implicit-int-float-conversion -Wno-tautological-overlap-compare -Wno-deprecated-copy -Wno-range-loop-construct -Wno-zero-as-null-pointer-constant -Wno-deprecated-anon-enum-enum-conversion -Wno-pessimizin
g-move -Wno-non-c-typedef-for-linkage -Wno-align-mismatch -Wno-error=unused-but-set-variable -Wno-error=unused-but-set-parameter -Wno-error=deprecated-builtins -Wno-error=deprecated -Wno-error=single-bit-bitfield-constant-conversion -Wno
-error=enum-constexpr-conversion -Wno-format-insufficient-args -Wno-sizeof-array-div -Wno-incompatible-function-pointer-types -Wno-unused-but-set-variable -Wno-unused-but-set-parameter -Wno-unqualified-std-cast-call -Wno-bitwise-instead-
of-logical -Wno-misleading-indentation -Wno-array-parameter -Wno-gnu-offsetof-extensions -MD -MF out/target/product/hammerhead/obj/SHARED_LIBRARIES/memtrack.msm8974_intermediates/memtrack_msm.d -o out/target/product/hammerhead/obj/SHARED
_LIBRARIES/memtrack.msm8974_intermediates/memtrack_msm.o hardware/qcom/display/msm8974/libmemtrack/memtrack_msm.c"
In file included from hardware/qcom/display/msm8974/libmemtrack/memtrack_msm.c:22:
In file included from hardware/libhardware/include/hardware/memtrack.h:24:
hardware/libhardware/include/hardware/hardware.h:23:10: fatal error: 'cutils/native_handle.h' file not found
#include <cutils/native_handle.h>

```

--- a/msm8974/libgralloc/Android.mk
+++ b/msm8974/libgralloc/Android.mk
@@ -27,7 +27,7 @@ LOCAL_MODULE_TAGS             := optional
 LOCAL_SHARED_LIBRARIES        := $(common_libs) libmemalloc
 LOCAL_SHARED_LIBRARIES        += libqdutils libGLESv1_CM
 LOCAL_CFLAGS                  := $(common_flags) -DLOG_TAG=\"qdgralloc\"
-LOCAL_HEADER_LIBRARIES        := display_headers generated_kernel_headers
+LOCAL_HEADER_LIBRARIES        := display_headers generated_kernel_headers libhardware_headers
 LOCAL_SRC_FILES               := gpu.cpp gralloc.cpp framebuffer.cpp mapper.cpp
 
 include $(BUILD_SHARED_LIBRARY)
@@ -40,7 +40,7 @@ LOCAL_PROPRIETARY_MODULE      := true
 LOCAL_MODULE_TAGS             := optional
 LOCAL_SHARED_LIBRARIES        := $(common_libs) libqdutils libdl
 LOCAL_CFLAGS                  := $(common_flags) -DLOG_TAG=\"qdmemalloc\"
-LOCAL_HEADER_LIBRARIES        := display_headers generated_kernel_headers
+LOCAL_HEADER_LIBRARIES        := display_headers generated_kernel_headers libhardware_headers
 LOCAL_SRC_FILES               := ionalloc.cpp alloc_controller.cpp
 
 include $(BUILD_SHARED_LIBRARY)

--- a/msm8974/libmemtrack/Android.mk
+++ b/msm8974/libmemtrack/Android.mk
@@ -20,7 +20,7 @@ include $(CLEAR_VARS)
 
 LOCAL_MODULE_RELATIVE_PATH    := hw
 LOCAL_PROPRIETARY_MODULE      := true
-LOCAL_C_INCLUDES += hardware/libhardware/include
+LOCAL_HEADER_LIBRARIES := libhardware_headers
 LOCAL_SHARED_LIBRARIES := liblog
 LOCAL_SRC_FILES := memtrack_msm.c kgsl.c
 LOCAL_MODULE := memtrack.$(TARGET_BOARD_PLATFORM)

--- a/msm8974/libqdutils/Android.mk
+++ b/msm8974/libqdutils/Android.mk
@@ -16,7 +16,7 @@ include $(BUILD_SHARED_LIBRARY)
 include $(CLEAR_VARS)
 
 LOCAL_SHARED_LIBRARIES          := liblog libcutils
-LOCAL_HEADER_LIBRARIES          := display_headers generated_kernel_headers
+LOCAL_HEADER_LIBRARIES          := display_headers generated_kernel_headers libhardware_headers
 LOCAL_SRC_FILES                 := qdMetaData.cpp
 LOCAL_CFLAGS                    := $(common_flags)
 LOCAL_CFLAGS                    += -DLOG_TAG=\"DisplayMetaData\"

